### PR TITLE
[iOS] Fix SearchResultScreen hidden bug

### PR DESCRIPTION
**Pull request by ostk0069**
## Issue
- close #ISSUE_NUMBER

## Overview (Required)
- In current latest `main` branch, there are bug in search result screen. So I fixed it with refactoring.
  How the bug appears
  1. show search result in Media tab
  2. switch bottom tab to other tab
  3. go back to media tab
  4. search result should be shown, but it does not.

## Links
-

## Screenshot
Before

https://user-images.githubusercontent.com/27538852/125171085-3eadac00-e1ed-11eb-8627-64140e7bdfc7.mov

After

https://user-images.githubusercontent.com/27538852/125171030-f4c4c600-e1ec-11eb-8a7b-d89351aa4e4c.mov

### CODE REVIEW

File: ios/Sources/MediaFeature/MediaListFeature.swift [hunk 1]
```diff
@@ -17,22 +17,28 @@ public struct MediaListState: Equatable {
     var blogs: [FeedContent]
     var videos: [FeedContent]
     var podcasts: [FeedContent]
-    var next: Next?
+    var isSearchResultVisible: Bool
+    var isSearchTextEditing: Bool
+    var isMoreActiveType: MediaType?
```
Comment (ry-itto) [inline ios/Sources/MediaFeature/MediaListFeature.swift:22]: I think `is**` is for condition variable name.
So how about rename this with `moreActiveType` ?

File: ios/Sources/MediaFeature/MediaListFeature.swift [hunk 2]
```diff
@@ -55,38 +61,21 @@ public enum MediaType {
...
-        case nil, .searchText, .isEditingDidChange:
-            state.next = searchText.map { .searchText($0) }
-            if let searchText = searchText {
-                state.searchedFeedContents = state.feedContents.filter { content in
-                    content.item.title.jaTitle.lowercased().contains(searchText.lowercased())
-                }
+        state.isSearchResultVisible = !(searchText?.isEmpty ?? true)
+        if let searchText = searchText {
+            state.searchedFeedContents = state.feedContents.filter { content in
+                content.item.title.jaTitle.lowercased().contains(searchText.lowercased())
```
Comment (ry-itto) [inline ios/Sources/MediaFeature/MediaListFeature.swift:67]: How about search with `enTitle` as well?

**Comment by ostk0069** [comment]
@ry-itto 
fixed 🙏

**Review by ry-itto — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**